addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js

```diff
@@ -79,68 +80,78 @@ odoo.define('point_of_sale.PartnerListScreen', function(require) {
             } else {
                 res = this.env.pos.db.get_partners_sorted(1000);
             }
-            return res.sort(function (a, b) { return (a.name || '').localeCompare(b.name || '') });
-        }
-        get isNextButtonVisible() {
-            return this.state.selectedPartner ? true : false;
-        }
-        /**
-         * Returns the text and command of the next button.
-         * The command field is used by the clickNext call.
-         */
-        get nextButton() {
-            if (!this.props.partner) {
-                return { command: 'set', text: this.env._t('Set Customer') };
-            } else if (this.props.partner && this.props.partner === this.state.selectedPartner) {
-                return { command: 'deselect', text: this.env._t('Deselect Customer') };
-            } else {
-                return { command: 'set', text: this.env._t('Change Customer') };
+            res.sort(function (a, b) { return (a.name || '').localeCompare(b.name || '') });
+            // the selected partner (if any) is displayed at the top of the list
+            if (this.state.selectedPartner) {
+                let indexOfSelectedPartner = res.findIndex( partner => 
+                    partner.id === this.state.selectedPartner.id
+                );
+                if (indexOfSelectedPartner !== -1) {
+                    res.splice(indexOfSelectedPartner, 1);
+                    res.unshift(this.state.selectedPartner);
+                }
             }
+            return res
```

Comment:
Based on the assigned value of `res` in the previous lines, it looks like the array is already sorted. This can be more efficient. Can you create a task to optimize this later? I don't think optimizing this is a coverage of this task.

Reply:
Hi Joseph, great point! Is there a built-in function that implements binary search? If so we could easily adjust that in the present task! What do you think?

Reply:
That is a great point! I will try to incorporate that in my current task, thanks!